index.js

```diff
@@ -13,25 +13,36 @@ module.exports = {
 
   included() {
     let app = findHost(this);
+    let desiredComponents = [
+      'apacheconf',
+      'bash',
+      'css',
+      'http',
+      'javascript',
+      'json',
+      'json5',
+      'markup-templating',
+      'ruby',
+      'scss',
+      'yaml',
+      'typescript',
+      'diff',
+    ];
+    const existingComponents = app.options['ember-prism']?.components ?? [];
+    const newComponents = existingComponents.slice();
+    for (const component of desiredComponents) {
+      if (!existingComponents.includes(component)) {
+        newComponents.push(component);
+      }
+    }
```

Comment:
If this logic is only needed if there _are_ `app.options['ember-prism']` then let's move it to the appropriate branch below.

Reply:
I had figured that, by ensuring the only significant block of code was always executed, we guaranteed it would always be well tested, especially where this code was in a difficult place to test independently of normal operation. I was killing risk, which is usually my first consideration, by exposing it to light. With only a dozen or so items, performance doesn't enter into it in any significant way.

However, I can move it around. Sure.